### PR TITLE
[5.5] Pluralize words with float counts not equal to 1

### LINKED ISSUE #21983: Str::plural() returns singular for decimal quantity between 1 and 2
- Laravel Version: 5.5.19

### Description:

`Str::plural()` returns singular when `1 < $count < 2`.

I proposed a solution in #21973 but it was suggested that wasn't the right fix.

### Steps To Reproduce:

```php
$days = 1.5;
echo $days . ' ' . str_plural('day', $days);
// Actual => "1.5 day"
// Expected => "1.5 days"
```

### PR DESCRIPTION
**Pull request by atanas-dev**
My implementation to resolve #21983.

Shamelessly taking use case example from #21973:
```php
$remaining = 1.3;
echo 'You have ' . $remaining . ' ' . str_plural('gigabyte', $remaining) . ' of space remaining on your account.';

// Current behavior:
>>> You have 1.3 gigabyte of space remaining on your account.

// After this PR:
>>> You have 1.3 gigabytes of space remaining on your account.
```

Casting to string eliminates the float inaccuracies that cause issues when comparing to integers. For example:
```php
$count = 1 + 0.001 - 0.001;
(int) $count === 1
    => false
(string) $count === '1'
    => true
```

Let me know if this is an acceptable solution and if a comment should be added to clarify why string casting is used.

### CODE REVIEW

**Comment by sisve** [comment]
This functionality is based on the default value of precision 14. This value is a finicky thing in php, it allows php to throw away precision to round numbers to what people expect them to be, just like you expect `1 + 0.001 - 0.001` to be equal to 1.

Use `ini_set('precision', 17);` to have php take into consideration all the bits in the floating point number. This means that `1 + 0.001 - 0.001` will become (or at least represented in var_dump) as `0.99999999999999989` and your comparison to '1' will fail.

In short, this way of doing it will not work for those with a higher than default floating point precision.

Related reading; https://www.leaseweb.com/labs/2013/06/the-php-floating-point-precision-is-wrong-by-default/

**Comment by taylorotwell** [comment]
Handle the rounding before you pass it in.

**Comment by atanas-angelov-dev** [comment]
@sisve I guess I was being quite naive about my approach.

Is this really solvable without specifying an arbitrary precision (possibly user-overridable) to round the passed float when comparing to 1?
